test/unit/LDAPPropertyTests.cs

```diff
@@ -832,5 +852,213 @@ public void LDAPPropertyProcessor_ParseAllProperties_CollectionCountOne_NotBadPa
+                {
+                    {"useraccountcontrol", 0x1000000.ToString()},
+                    {
+                        "msds-allowedtodelegateto", new[]
+                        {
+                            "ldap/PRIMARY.testlab.local/testlab.local",
+                            "ldap/PRIMARY.testlab.local",
+                            "ldap/PRIMARY",
+                            "ldap/WIN10"
+                        }
+                    }
+                }, "S-1-5-21-3130019616-2776909439-2417379446-1101", Label.Computer);
+
+            var utils = new MockLDAPUtils();
+            var processor = new LDAPPropertyProcessor(utils);
+            var props = await processor.ReadPropertyDelegates(mock);
+            var delegates = props.Select(d => d.ObjectIdentifier);
+            
+            foreach (var principal in mock.GetArrayProperty("msds-allowedtodelegateto"))
+            {
+                var host = await utils.ResolveHostToSid(principal, mock.GetProperty(LDAPProperties.DistinguishedName));
+                Assert.Single(delegates, host);
+            }
+        }
+
+        [WindowsOnlyFact]
+        public void LDAPPropertyProcessor_ReadSidHistory_ReturnsPopulatedList()
+        {
+            var mock = new MockSearchResultEntry("CN\u003dWIN10,OU\u003dTestOU,DC\u003dtestlab,DC\u003dlocal",
+                new Dictionary<string, object>
+                {
+                    {
+                        "sidhistory", new[]
+                        {
+                            Helpers.B64ToBytes("AQUAAAAAAAUVAAAAIE+Qun9GhKV2SBaQUQQAAA==")
+                        }
+                    },
+                }, "S-1-5-21-3130019616-2776909439-2417379446-1101", Label.Computer);
+            
+            var processor = new LDAPPropertyProcessor(new MockLDAPUtils());
+            var sids = processor.ReadSidHistory(mock);
+            
+            Assert.Contains("S-1-5-21-3130019616-2776909439-2417379446-1105", sids);
+            Assert.Single(sids);
+        }
+
+        [Fact]
+        public void LDAPPropertyProcessor_ReadSidPrincipal_GetPrincipal()
+        {
+            var mock = new MockSearchResultEntry("CN\u003dWIN10,OU\u003dTestOU,DC\u003dtestlab,DC\u003dlocal",
+                new Dictionary<string, object>(), "S-1-5-21-3130019616-2776909439-2417379446-1101", Label.Computer);
+            
+            var sid = "S-1-5-21-3130019616-2776909439-2417379446-1105";
+            var processor = new LDAPPropertyProcessor(new MockLDAPUtils());
+            var principal = processor.ReadSidPrincipal(mock, sid);
+            
+            Assert.Equal(new TypedPrincipal
+            {
+                ObjectIdentifier = "S-1-5-21-3130019616-2776909439-2417379446-1105",
+                ObjectType = Label.User
+            }, principal);
+        }
+
+        // [Fact]
+        // public void LDAPPropertyProcessor_ReadAllowedToActPrincipals_ReturnsPopulatedList()
+        // {
```

Comment:
This may be a limitation of platform, need to test further, but failed to produce mocks for ActiveDirectorySecurityDescriptor for this test.  Will remove before merging if this can't be resolved.